Adyen/Core/Components/Base/Component.swift

```diff
@@ -14,10 +14,19 @@ extension Component {
 
     /// Finalizes the payment if there is any, after being processed by payment provider.
     /// - Parameter success: The status of the payment.
-    /// :nodoc:
+    @available(*, deprecated, message: "Use finalizeIfNeeded(with:, completion:) instead.")
     public func finalizeIfNeeded(with success: Bool) {
-        (self as? FinalizableComponent)?.didFinalize(with: success)
         stopLoadingIfNeeded()
```

Comment:
if  this deprecated version is used, the `stopLoadingIfNeeded()` will be called twice here and in the new `version`, maybe remove `stopLoadingIfNeeded()` from here since its called in the new version anyway?

Reply:
Good one!